Frontend/ui-driver/src/Screens/HomeScreen/View.purs

```diff
@@ -2342,25 +2343,53 @@ getDriverStatusResult index driverStatus currentStatus= case (getValueToLocalSto
                   _ -> if (driverStatus == currentStatus) then ACTIVE
                        else DEFAULT
 
-playAudioAndLaunchMap :: forall action. (action -> Effect Unit) ->  action -> (String -> action) -> Boolean -> Maybe String -> Number -> Flow GlobalState Unit
-playAudioAndLaunchMap push action audioCompleted acRide requestedVehicleVariant tollCharges = do
+geVehicleBasedAudioConfig :: RideStartAudio -> String -> Boolean -> StartAudioUrls
+geVehicleBasedAudioConfig rideStartAudio vehicleType isAcRide =
+  let audioConfig = if isAcRide then
+                      rideStartAudio.acCab
+                    else
+                      case vehicleType of
+                        "AUTO_RICKSHAW" -> rideStartAudio.auto
+                        "BIKE" -> rideStartAudio.bike
+                        _ -> rideStartAudio.nonAcCab
+  in
+    audioConfig
+  
+extractAudioConfig :: HomeScreenState -> Maybe String -> Boolean -> Maybe String -> Number -> Maybe String
+extractAudioConfig state requestedVehicleVariant acRide locationTag tollCharges =
+  let
+    isAcRide = fromMaybe false state.data.activeRide.acRide
+    vehicleType = fromMaybe "" state.data.activeRide.requestedVehicleVariant
+    isAirportPriority = DS.contains(DS.Pattern("SureAirport")) (fromMaybe "" locationTag)
+    cityConfig = HU.getCityConfig state.data.config.cityConfig (getValueToLocalStore DRIVER_LOCATION)
+    vehicleAudioConfig = geVehicleBasedAudioConfig cityConfig.rideStartAudio vehicleType acRide
+
+    audioUrl = if isAirportPriority && isJust vehicleAudioConfig.parkingAudio then
+                 vehicleAudioConfig.parkingAudio
+               else if tollCharges > 0.0 && isJust vehicleAudioConfig.tollAudio then
+                 vehicleAudioConfig.tollAudio
+               else if isJust vehicleAudioConfig.acAudio then
+                 spy "vehicleAudioConfig.acAudio" vehicleAudioConfig.acAudio
```

Comment:
We can remove the spy

Reply:
return directly instead of storing in audio URL and return it

Reply:
Done